### PR TITLE
Use GetLocation in more places

### PR DESCRIPTION
**Pull request by ZehMatt**

### CODE REVIEW

File: src/openrct2/paint/sprite/Paint.Sprite.cpp [hunk 1]
```diff
@@ -65,21 +65,23 @@ void sprite_paint_setup(paint_session* session, const CoordsXY& pos)
             }
         }
 
+        const auto entityPos = spr->GetLocation();
```
Comment (Gymnasiast) [inline src/openrct2/paint/sprite/Paint.Sprite.cpp:68]: Just checking: not `const auto&`?

File: src/openrct2/peep/Guest.cpp [hunk 2]
```diff
@@ -5027,11 +5027,13 @@ void Guest::UpdateRideShopLeave()
 {
     if (auto loc = UpdateAction(); loc.has_value())
     {
-        MoveTo({ loc.value(), z });
+        const auto curLoc = GetLocation();
+
+        MoveTo({ loc.value(), curLoc.z });
 
-        if ((x & 0xFFE0) != NextLoc.x)
+        if ((curLoc.x & 0xFFE0) != NextLoc.x)
```
Comment (ZehMatt) [inline src/openrct2/peep/Guest.cpp:5034]: This needs to use the new location

Comment (Gymnasiast) [inline src/openrct2/peep/Guest.cpp:5034]: This looks like a hand-rolled `ToTileStart()`

File: src/openrct2/paint/sprite/Paint.Sprite.cpp (hunk 1 above)
Comment (ZehMatt) [reply src/openrct2/paint/sprite/Paint.Sprite.cpp:68]: GetLocation currently returns a copy as we are not storing that just yet as CoordsXYZ

File: src/openrct2/peep/Peep.cpp [hunk 3]
```diff
@@ -474,13 +474,14 @@ std::optional<CoordsXY> Peep::UpdateAction(int16_t& xy_distance)
     WindowInvalidateFlags |= PEEP_INVALIDATE_PEEP_2;
 
     // Create sick at location
```
Comment (Gymnasiast) [inline src/openrct2/peep/Peep.cpp:476]:
```suggestion
```

It’s clear from the code anyway, and it’s a colloquial usage of the word that won’t be familiar to everyone.

**Review by Gymnasiast — APPROVED** [review]

**Pull request merged**